contracts/NonfungiblePositionManager.sol

```diff
@@ -90,26 +88,14 @@ contract NonfungiblePositionManager is INonfungiblePositionManager, BaseLiquidit
         uint256 deadline,
         address owner,
         bytes calldata hookData
-    ) public payable returns (BalanceDelta delta) {
-        // TODO: optimization, read/write manager.isUnlocked to avoid repeated external calls for batched execution
-        if (manager.isUnlocked()) {
```

Comment:
I actually think all of these functions should have a onlyIfUnlocked modifer that checks a variable we transiently store at the beginning of `unlockAndExecute`

```
modifer onlyIfUnlocked() {

if (locked) revert MustBeUnlockedByThisContract();

}
```